impl/src/java/org/sakaiproject/evaluation/logic/externals/EvalExternalLogicImpl.java

```
@@ -480,55 +482,44 @@ public EvalGroup makeEvalGroupObject(String evalGroupId) {
+        try {
+            // try to get the site object based on the entity reference (which is the evalGroupId)
+            // first we try to go straight to the siteService which is fastest
+            if (evalGroupId.contains(GROUP_PREFIX)) {
+                groupTitle = siteService.findGroup(evalGroupId).getTitle();
+                groupType = EvalConstants.GROUP_TYPE_GROUP;
+            } else if (evalGroupId.startsWith(SITE_PREFIX)) {
+                try {
+                    String siteId = evalGroupId.substring(SITE_PREFIX.length());
+                    groupTitle = siteService.getSite(siteId).getTitle();
+                    groupType = EvalConstants.GROUP_TYPE_SITE;
+                } catch (IdUnusedException e) {
+                    log.warn("Exception caught while trying to get site", e);
                 }
-            } catch (Exception e) {
-                // invalid site reference
-                log.debug("Could not get sakai site from evalGroupId:" + evalGroupId, e);
-                c = null;
             }
-        }
 
-        if (c == null) {
-            log.error("Could not get group from evalGroupId:" + evalGroupId);
-            // create a fake group placeholder as an error notice
-            c = new EvalGroup( evalGroupId, "** INVALID: "+evalGroupId+" **", 
-                    EvalConstants.GROUP_TYPE_INVALID );
+            // next try getting from entity system
+            if (groupTitle == null) {
+                Object entity = entityBroker.fetchEntity(evalGroupId);
+                if (entity instanceof Site) {
+                    groupTitle = ((Site) entity).getTitle();
+                    groupType = EvalConstants.GROUP_TYPE_SITE;
+                } else if (entity instanceof Group) {
+                    groupTitle = ((Group) entity).getTitle();
+                    groupType = EvalConstants.GROUP_TYPE_GROUP;
+                } else {
+                    throw new IllegalArgumentException("Invalid group id");
```

I don't want to change the code too much, but it would have made sense to not catch that one anyways.
We can see that the first condition in this method throws an `IllegalArgumentException` so it's handled by the calling code (I assume), and it would be logic to throw that exception when the given ID is invalid (instead of creating a fake group).